src/core/trulens/experimental/otel_tracing/core/instrument.py

```diff
@@ -120,8 +128,13 @@ def wrapper(*args, **kwargs):
                             )
 
                 except Exception as e:
+                    attributes_exception = e
                     logger.error(f"Error setting attributes: {e}")
-                    return None
+
+                if func_exception:
+                    raise func_exception
+                if attributes_exception:
+                    raise attributes_exception
```

Comment:
just a small note - raising an exception here will cause this to be recorded as a span with an error, I was thinking about it and wasn't sure if that was what we wanted for an attributes exception, but I could see it going either way/

Reply:
I think it makes sense if the span has an error because if we had an error during the attribute computation, the span wouldn't be filled in properly so it'd be good to know that.